src/core/js/helpers.js

```diff
@@ -101,11 +101,37 @@ define([
 
         /**
          * makes the _globals object in course.json available to a template
-         */ 
+         */
         import_globals: function(context) {
-            if(!context.data.root._globals) {
+            if (!context.data.root._globals) {
                 context.data.root._globals = Adapt.course.get('_globals');
             }
+            return "";
+        },
+
+        /**
+         * makes the course data available to a template
+         */
+        import_data: function(context) {
+            
+            var directImport = ['config','course'];
+            for (var i = 0, l = directImport.length; i < l; i++) {
+                var name = directImport[i];
+                context.data.root['_'+name] = Adapt[name].toJSON();
+            }
+
+            var indexedImport = ['contentObjects','articles','blocks','components'];
+            for (var i = 0, l = indexedImport.length; i < l; i++) {
```

Comment:
I think you could switch to `for...in` for both these `for` loops? or `_.each`?

Reply:
although if you want this to go into Adapt v2.x you'll need to include a `hasOwnProperty` check with the `for...in` to stop it from iterating over the methods that the ES5 Shim adds to Array... might be an idea to include that regardless.

Reply:
I've replaced ```forEach``` calls for ```for``` loops after Tom G cited performance concerns as this could be used in every ```component.hbs``` template and would need to remap the course data every time.

Reply:
```for...in``` is an odd type of loop to use on an Array. It's an enumerable key loop, which I guess would work fine with arrays but isn't ... really... proper...?

Reply:
I don't see why it's not 'proper', in fact the MDN docs refer specifically to using it on an Array! So long as you don't mind that it won't iterate through the Array in a specific order and you do a `hasOwnProperty` check it works great, nice tidy syntax - esp. when you've got an inner loop too.

Reply:
mmmm so tidy 
```javascript
var directImport = ['config','course'];
for (let name in directImport) {
    if(directImport.hasOwnProperty(name)) {
        context.data.root['_'+name] = Adapt[name].toJSON();
    }
}

var indexedImport = ['contentObjects','articles','blocks','components'];
for (let name in indexedImport) {
    if (!indexedImport.hasOwnProperty(name)) continue;
    var importArray = Adapt[name].toJSON();
    var importIndex = {};
    for (let item in importArray) {
        if(importArray.hasOwnProperty(item)) {
            importIndex[item._id] = item;
        }
    }
    context.data.root['_' + name] = importIndex;
}
```

Reply:
I mean that the intention of a ``for...in`` is to iterate through an Object's enumerable properties and not an Array's indices. Even though those two ideas have overlap and are technically similar in that they can both be enumerable properties, if I see a ``for...in`` my assumption is that the code is acting upon and Object's name value pairs and not an Array's indices. To me it would feel as though the code obscures the intention of the loop for the sake of neater syntax. 
As the ``for...in`` doesn't restrict itself to only the numeric indices from the Object's property names, if the Array was extended or had an extra enumerable property for some reason, the ``for...in`` would create unnecessary addition logic problems to solve without an easy or clear solution, even with a ``hasOwnProperty`` check. 
Adding the ```hasOwnProperty``` check means that I have to include an extra function call for each indices. Which kind of negates the point of switching from a ``forEach``.
From my perspective using a ``for...in`` makes the code unnecessary fragile, less performant and more obtuse for very little benefit.

Reply:
Although you are looping over the array indices in your code, in this instance it's only for the purposes of getting hold of the array values, which a `for...in` loop does for you

Unnecessarily fragile? Not with the `hasOwnProperty` check. In what circumstances would that not work?

More obtuse? Very subjective, that. Reads a LOT clearer to me. We'll have to agree to disagree on that point I fear.

Less performant? Yes, possibly. Will the impact be more than negligible though? 

"The real problem is that programmers have spent far too much time worrying about efficiency in the wrong places and at the wrong times; premature optimization is the root of all evil (or at least most of it) in programming." - Donald Knuth

Anyway it's all moot as these are just suggestions, happy for it to go in as-is

Reply:
```js
var directImport = ['config','course'];
directImport.prop = "a";
for (let name in directImport) {
    console.log(name);
}
```

Reply:
> I don't understand why you're intentionally ignoring this point

Because I don't agree that it's an established pattern of use

> Moving from the `forEach` is not premature optimization

I wasn't referring to that.

I'm sure you'll see I have +1'd this so I think we need to agree to disagree and move on!

Reply:
Fair enough.